src/components/ScrollLock/index.js

```diff
@@ -1,35 +1,82 @@
 import React, {PureComponent as Component} from 'react'
 import PropTypes from 'prop-types'
+import { noop } from '../../utilities/other'
 
 export const propTypes = {
-  isDisabled: PropTypes.bool
+  isDisabled: PropTypes.bool,
+  direction: PropTypes.oneOf(['x', 'y']),
+  stopPropagation: PropTypes.bool,
+  onWheel: PropTypes.func
 }
 const defaultProps = {
-  isDisabled: false
+  isDisabled: false,
+  direction: 'y',
+  stopPropagation: false,
+  onWheel: noop
 }
 
 class ScrollLock extends Component {
   render () {
-    const { children, isDisabled } = this.props
+    const {
+      children,
+      direction,
+      isDisabled,
+      onWheel,
+      stopPropagation
+    } = this.props
 
     if (!children) {
       return null
     }
 
     const child = React.Children.only(children)
     const events = {
-      onWheel: handleWheelEvent
+      onWheel: event => {
+        handleWheelEvent(event, direction, stopPropagation)
+        onWheel(event)
```

Comment:
Do you think these calls should be throttled?

Reply:
🤔 I think in the case of ScrollLock (and Overflow), we **can't** throttle the scroll event, as it would result in a scroll-jacking vibe. The calculations need to fire at every tick. But I totally know where your concern is coming from.

I've done extensive testing to check for re-draw performance and fps drops. So far, these components hold up very well, and fps stays at 60 with low CPU/GPU load.

I think we'll only need to consider throttling if we notice dropped frames or jankiness.

Hope that makes sense!!

Reply:
Sounds good!